reporting/src/offscreen/doublefetch/index.js

```diff
@@ -0,0 +1,38 @@
+(function () {
+  const domReady = new Promise((resolve) => {
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', () => {
+        resolve();
+      });
+    } else {
+      resolve();
+    }
+  });
+
+  chrome.runtime.onMessage.addListener(
+    async (message, sender, sendResponse) => {
```

Comment:
please avoid async onMessage listeners, they always return the promise which if truthy value and will make the listener wait until sendResponse is executed